crates/biome_cli/src/execute/migrate/eslint_eslint.rs

```diff
@@ -364,10 +364,8 @@ impl<T: Deserializable + 'static, U: Deserializable + 'static> Deserializable fo
                 Some(RuleConf::Spread(severity, spread))
             }
         }
-        if matches!(
-            value.visitable_type()?,
-            VisitableType::NUMBER | VisitableType::STR
-        ) {
+        let visitable_type = value.visitable_type()?;
+        if visitable_type == VisitableType::NUMBER || visitable_type == VisitableType::STR {
```

Comment:
I have mixed feeling here. This doesn't look right to return a `VisitableType` when a value is expected to have a single type.
I think we should change the return type of `DeserializableValue::visitable_type` to `Types`.

This change also requires the signature of `DeserializationDiagnostic::new_incorrect_type_with_name` and `DeserializationDiagnostic::new_incorrect_type_with_name`: `actual_type` should be a `Types`.